src/mongo/db/storage/kv/kv_catalog.cpp

```diff
@@ -224,7 +224,9 @@ BSONObj KVCatalog::_findEntry(OperationContext* opCtx, StringData ns, RecordId*
 
     LOG(3) << "looking up metadata for: " << ns << " @ " << dl;
     RecordData data;
-    if (!_rs->findRecord(opCtx, dl, &data, skipPessimisticLocking)) {
+    const bool found = skipPessimisticLocking ? _rs->findRecordRelaxed(opCtx, dl, &data)
```

Comment:
Can we break these two calls into two separate blocks of an if/else statement?  I would prefer, for both legibility and debugging, that we not use the ternary operator here.  Something like:

```
bool recordWasFound = false;
if (skipPessimisticLocking) {
  recordWasFound = _rs->findRecordRelaxed(opCtx, dl, &data);
} else {
  recordWasFound = _rs->findRecord(opCtx, dl, &data);
}